lib/stages/VisSharedMemWriter.cpp

```diff
@@ -121,10 +123,14 @@ void VisSharedMemWriter::wait_for_semaphore() {
     }
 
     ts.tv_sec += _wait_time;
+    double start_time = current_time();
+
     if (sem_timedwait(sem, &ts) == -1) {
         FATAL_ERROR("sem_timedwait() timed out");
         return;
     }
+
+    access_record_wait_time_seconds.labels({_name}).set(current_time() - start_time);
```

Comment:
We can't guarantee, the metric is scraped every time it changes in this line. So wouldn't it make more sense to accumulate the time?

Reply:
If it is a total accumulated wait time, how do we assess which baseline time is worth our attention?